### PR TITLE
[homekit] support actual booleans in metadata config

### PR DESCRIPTION
**Pull request by ccutrer**
as well as strings, too.

reduces confusion, if someone doesn't know it needed to be a string

### CODE REVIEW

**Comment by ccutrer** [comment]
Don't bother reviewing yet... I need to push up another commit that applies it to other booleans

**Comment by ccutrer** [comment]
okay, it's better now

**Review by yfre — APPROVED** [review]
LGTM

**Comment by yfre** [comment]
@ccutrer thank you for this fix. 
it was indeed confusing, even i lost few hours myself trying to understand why "inverted = true" is not working :)

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitTaggedItem.java [hunk 1]
```diff
@@ -211,14 +211,38 @@ public <T> T getConfiguration(String key, T defaultValue) {
         return defaultValue;
     }
 
+    /**
+     * return configuration as booelan if exists otherwise return defaultValue
```
Comment (lolodomo) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitTaggedItem.java:215]: Typo: boolean
And "**Returns** configuration **value** as..."

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitTaggedItem.java [hunk 2]
```diff
@@ -211,14 +211,38 @@ public <T> T getConfiguration(String key, T defaultValue) {
...
+     *
+     * @param key configuration key
+     * @param defaultValue default value
+     * @return value
```
Comment (lolodomo) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitTaggedItem.java:219]: `@return configuration value as boolean`

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**